lib/core/application/chat/check_guild_room_roles_bloc/check_guild_room_roles_bloc.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:app/core/domain/chat/chat_repository.dart';
+import 'package:app/core/domain/chat/entities/guild.dart';
+import 'package:app/core/domain/chat/entities/guild_room.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'check_guild_room_roles_bloc.freezed.dart';
+
+class CheckGuildRoomRolesBloc
+    extends Bloc<CheckGuildRoomRolesEvent, CheckGuildRoomRolesState> {
+  final _chatRepository = getIt<ChatRepository>();
+
+  final GuildRoom guildRoom;
+
+  CheckGuildRoomRolesBloc({required this.guildRoom})
+      : super(CheckGuildRoomRolesStateLoading()) {
+    on<CheckGuildRoomRolesEventFetch>(_onFetch);
+  }
+
+  Future<void> _onFetch(
+    CheckGuildRoomRolesEventFetch event,
+    Emitter emit,
+  ) async {
+    emit(CheckGuildRoomRolesState.loading());
+    final guild = await _getGuildDetail();
+    final guildRolePermissions =
+        await _getGuildRolePermissions(event.walletAddress);
+    final guildRoleIds = guildRoom.guildRoleIds;
+    List<GuildRole> filteredRoles = [];
+    if (guild != null) {
+      if (guildRoleIds != null) {
+        filteredRoles = guild.roles!
```

Comment:
how you make sure `guild.roles`wil not be null ? Please be careful when using `!`, otherwise please use [] as a fallback value